picamera2/picamera2.py

```diff
@@ -235,6 +235,7 @@ def preview_configuration(self, main={}, lores=None, raw=None, transform=libcame
         lores = self.make_initial_stream_config({"format": "YUV420", "size": main["size"]}, lores)
         raw = self.make_initial_stream_config({"format": self.sensor_format, "size": main["size"]}, raw)
         controls = {"NoiseReductionMode": 3} | controls
+        self.set_controls(controls)
```

Comment:
I think this might be a problem. The intention is that while the camera is running you can create another configuration (for example, call still_configuration() while the preview is running). I'm guessing this might cause the control values intended for the new mode to be sent during the preview?

Reply:
That would be correct. There are two options I think could work.

1) Remove set_controls from prebuilt configurations and move it to .configure so the controls aren't changed until the user submits configure.
2) Remove the controls options from the prebuilts. Separate control setting from stream configuration. I don't think this would difficult to conceptualize with the right documentation.

Reply:
I think 1 sounds OK. There are specific reasons why particular configurations "recommend" particular control values, and I wouldn't want to do anything that makes it easier for users to end up with inappropriate choices by forgetting to do something.